pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -390,6 +390,21 @@ func (m *ManagerImpl) Allocate(pod *v1.Pod, container *v1.Container) error {
 
 }
 
+//@klueska I'm not sure that it is sufficient to revert resource allocation in device manager.
+func (m *ManagerImpl) DeAllocate(pod *v1.Pod, container *v1.Container) error {
```

Comment:
AFAIK there were some past attempt to implement this, and  no one materialized, see for example: https://github.com/kubernetes/kubernetes/issues/59110

Reply:
It seems that the discussion is to add DeAllocation API on **DevicePlugin API**.
I hope that DeAllocate function of hintprovider would work without Device Plugin API support.